packages/frontend/src/components/common/Filters/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { Button, Tag } from '@blueprintjs/core';
+import {
+    fieldId,
+    FilterableDimension,
+    FilterOperator,
+    FilterRule,
+    Filters,
+    getTotalFilterRules,
+    isFilterRule,
+} from 'common';
+import React, { FC, useCallback } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import FilterRuleForm from './FilterRuleForm';
+
+type Props = {
+    fields: FilterableDimension[];
+    filters: Filters;
+    setFilters: (value: Filters) => void;
+};
+
+const FiltersForm: FC<Props> = ({ fields, filters, setFilters }) => {
+    const rules: Array<FilterRule> = getTotalFilterRules(filters);
+
+    const onDeleteFilterRule = useCallback(
+        (index: number) => {
+            setFilters({
+                dimensions: {
+                    id: uuidv4(),
+                    and: [...rules.slice(0, index), ...rules.slice(index + 1)],
+                },
+            });
+        },
+        [rules, setFilters],
+    );
+
+    const onChangeFilterRule = useCallback(
+        (index: number, filterRule: FilterRule) => {
+            setFilters({
+                dimensions: {
+                    id: uuidv4(),
+                    and: [
+                        ...rules.slice(0, index),
+                        filterRule,
+                        ...rules.slice(index + 1),
+                    ],
+                },
+            });
+        },
+        [rules, setFilters],
+    );
+
+    const onAddFilterRule = useCallback(() => {
+        if (fields.length > 0) {
+            setFilters({
+                dimensions: {
+                    id: uuidv4(),
+                    and: [
+                        ...rules,
+                        {
+                            id: uuidv4(),
+                            target: {
+                                fieldId: fieldId(fields[0]),
+                            },
+                            operator: FilterOperator.NULL,
+                        },
+                    ],
+                },
+            });
+        }
+    }, [fields, rules, setFilters]);
+
+    return (
+        <div
+            style={{
+                margin: '10px',
+                display: 'flex',
+                flexDirection: 'column',
+                justifyContent: 'flex-start',
+                alignItems: 'start',
+            }}
+        >
+            <div
+                style={{
+                    width: '100%',
+                    paddingBottom: '20px',
+                    display: 'flex',
+                    flexDirection: 'column',
+                    gap: '5px',
+                }}
+            >
+                {rules.length > 0 && (
+                    <div
+                        style={{ display: 'inline-flex', alignItems: 'center' }}
+                    >
+                        <p style={{ margin: 0, marginRight: 10 }}>
+                            Operator: <Tag minimal>AND</Tag>
```

Comment:
This feels irrelevant when I've only got one filter:
<img width="636" alt="Screenshot 2022-01-31 at 15 59 29" src="https://user-images.githubusercontent.com/11660098/151828125-352c51ab-9e58-46b3-a494-f2fabe25e76c.png">

Reply:
<img width="1025" alt="Screenshot 2022-01-31 at 16 15 11" src="https://user-images.githubusercontent.com/11660098/151830868-af34a3b8-95e3-4769-87be-8f06bacc7588.png">

Alternative could be to use a left margin for this information

Reply:
I was trying to follow the same UI as this [library](https://react-querybuilder.js.org/react-querybuilder/#addRuleToNewGroups=false&autoSelectField=true&disabled=false&enableDragAndDrop=false&independentCombinators=false&resetOnFieldChange=true&resetOnOperatorChange=false&showCloneButtons=false&showCombinatorsBetweenRules=false&showNotToggle=false&validateQuery=false) 
In my next PR I'm going to make that label a dropdown to change between AND and OR. So it makes sense to only have it in 1 place. Open to suggestions.

Reply:
<img width="994" alt="Screenshot 2022-01-31 at 16 57 35" src="https://user-images.githubusercontent.com/11660098/151838047-fb9fd583-be77-4637-88c2-71d4467ce573.png">

I love this for inspiration! 

I think the placement at the side makes sense (vs. the top). Also in this example it avoids the problem of having many "AND"s if you want a single place to change the type

Reply:
But this is for another PR right?

Reply:
cc: @PriPatel